more security considerations (#24)
ietf-scitt/draft-birkholz-cose-tsa-tst-header-parameter: draft-birkholz-cose-tsa-tst-header-parameter.md

Also review the Security Considerations section in {{-COSE}}; these considerations apply to this document as well, especially the need for implementations to protect private key material.

The following scenario assumes an attacker can manipulate the clocks on the COSE signer and its relying parties, but not the TSA.
It is also assumed that the TSA is a trusted third party, so the attacker cannot impersonate the TSA and create valid timestamp tokens.
In such a setting, any tampering with the COSE signer's clock does not have an impact because, once the timestamp is obtained from the TSA, it becomes the only reliable source of time.

> However, in both CTT and TTC mode, a denial of service can occur if the attacker can adjust the relying party's clock to a past time.

Author: ```suggestion
However, in both CTT and TTC mode, a denial of service can occur if the attacker can adjust the relying party's clock to a future time.
```

> However, in both CTT and TTC mode, a denial of service can occur if the attacker can adjust the relying party's clock to a future time.

Author: ```suggestion
However, in both CTT and TTC mode, a denial of service can occur if the attacker can adjust the relying party's clock to a past time.
```

> However, in both CTT and TTC mode, a denial of service can occur if the attacker can adjust the relying party's clock to a future time.

Author: ```suggestion
However, in both CTT and TTC mode, a denial of service can occur if the attacker can adjust the relying party's clock so that the CMS validation fails.
```

> 

Author: ```suggestion
```

In CTT mode, an attacker could manipulate the unprotected header by removing or replacing the timestamp.

> To avoid that, the signed COSE object should be securely wrapped in an envelope during transit and at rest.

Author: ```suggestion
To avoid that, the signed COSE object should be integrity protected during transit and at rest.
```

Also review the Security Considerations section in {{-COSE}}; these considerations apply to this document as well, especially the need for implementations to protect private key material.

> We assume an attacker who has the ability to manipulate the clocks on the COSE signer's clock and its relying parties, but not the clock of the TSA.

Author: ```suggestion
We assume an attacker who has the ability to manipulate the clocks on the COSE signer and its relying parties, but not the clock of the TSA.
```

Also review the Security Considerations section in {{-COSE}}; these considerations apply to this document as well, especially the need for implementations to protect private key material.

> We assume an attacker who has the ability to manipulate the clocks on the COSE signer's clock and its relying parties, but not the clock of the TSA.

Reviewer: ```suggestion
Attackers could achieve the ability to manipulate the local clocks used by the COSE signer or local clocks of its relying parties, but not the clock of the TSA.
```

Author: the spirit of this suggestion has been captured in [e75174a](https://github.com/ietf-scitt/draft-birkholz-cose-tsa-tst-header-parameter/pull/24/commits/e75174a9281914f13bbd31b6287ad4e029991169)

> Additionally, we are assuming that the TSA is a trusted third party, meaning that the attacker cannot impersonate the TSA and create valid timestamp tokens.

Reviewer: ```suggestion
Additionally, the TSA is assumed to be a trusted third party, meaning that the attacker cannot impersonate the TSA and therefore the TSA can only create valid timestamp tokens.
```

Author: ditto
